contracts/v2/artist-controls/ArtistEditionBurner.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+interface IKODAV2ArtistBurner {
+  function editionActive(uint256 _editionNumber) external view returns (bool);
+
+  function artistCommission(uint256 _editionNumber) external view returns (address _artistAccount, uint256 _artistCommission);
+
+  function updateActive(uint256 _editionNumber, bool _active) external;
+
+  function totalSupplyEdition(uint256 _editionNumber) external view returns (uint256);
+
+  function totalRemaining(uint256 _editionNumber) external view returns (uint256);
+
+  function updateTotalAvailable(uint256 _editionNumber, uint256 _totalAvailable) external;
+}
+
+/**
+* @title Artists burning contract for KnownOrigin (KODA)
+*
+* Allows for the edition artists to burn unsold editions or reduce the supply of sold ones
+*
+* https://www.knownorigin.io/
+*
+* BE ORIGINAL. BUY ORIGINAL.
+*/
+contract ArtistEditionBurner is Ownable, Pausable {
+  using SafeMath for uint256;
+
+  // Interface into the KODA world
+  IKODAV2ArtistBurner public kodaAddress;
+
+  event EditionDeactivated(
+    uint256 indexed _editionNumber
+  );
+
+  event EditionSupplyReduced(
+    uint256 indexed _editionNumber
+  );
+
+  constructor(IKODAV2ArtistBurner _kodaAddress) public {
+    kodaAddress = _kodaAddress;
+  }
+
+  /**
+   * @dev Sets the provided edition to either a deactivated state or reduces the available supply to zero
+   * @dev Only callable from edition artists defined in KODA NFT contract
+   * @dev Only callable when contract is not paused
+   * @dev Reverts if edition is invalid
+   * @dev Reverts if edition is not active in KDOA NFT contract
+   */
+  function deactivateOrReduceEditionSupply(uint256 _editionNumber) external whenNotPaused {
+    (address artistAccount, uint256 _) = kodaAddress.artistCommission(_editionNumber);
+    require(msg.sender == artistAccount || msg.sender == owner, "Only from the edition artist account");
```

Comment:
Should the message mention owner? We are presuming artist should do it?

Reply:
Ye presumed artist control - owner override as last resort